taichi/runtime/llvm/runtime_module/node_dynamic.h

```diff
@@ -76,14 +77,12 @@ i32 Dynamic_append(Ptr meta_, Ptr node_, i32 data) {
       });
     }
     if (i < chunk_start + chunk_size) {
-      *(i32 *)(*p_chunk_ptr + sizeof(Ptr) +
-               (i - chunk_start) * meta->element_size) = data;
-      break;
+      return *p_chunk_ptr + sizeof(Ptr) +
+             (i - chunk_start) * meta->element_size;
     }
     p_chunk_ptr = (Ptr *)(*p_chunk_ptr);
     chunk_start += chunk_size;
   }
-  return i;
```

Comment:
oops, can we return some nullptr here and add a comment `NEVER REACH HERE` so that it's easier to read?